addons/mail/static/src/core/web/thread_model_patch.js

```diff
@@ -5,9 +5,21 @@ import { Thread } from "@mail/core/common/thread_model";
 import { patch } from "@web/core/utils/patch";
 
 patch(Thread.prototype, "mail/core/web", {
+    /** @type {integer|undefined} */
+    recipientsCount: undefined,
+    /** @type {Set<import("@mail/core/common/follower_model").Follower>|undefined} */
+    recipients: undefined,
+    /** @type {Number} */
+    mt_comment_id: undefined,
+    get recipientsFullyLoaded() {
+        return this.recipientsCount === this.recipients.size;
+    },
     /**
      * @returns {import("@mail/core/web/activity_model").Activity[]}
      */
+    setup() {
+        this.recipients = new Set();
+    },
```

Comment:
I was asked to move them in /web https://github.com/odoo/odoo/pull/130709#discussion_r1309046001

Reply:
Putting bits in different files and folders has a significant cost on code maintainability.
I'm okay to pay the price when we really need it, e.g. discuss client action requires some `web/` code.

In this case, this is just taking the ideological trend of over-encapsulating bits of data per environment too far: it forces you to add more LOCs (add `setup()` function), it breaks locality of concern of listing all fields of model (they are spread in many different files, hard to see all of them), and it makes this code much harder to reuse in other environments like portal chatter that we are currently working on.
This code splitting is also more error-prone, because the field is not listed with `@type {Thread}`.

cc @seb-odoo

Reply:
Locality of concern is actually much better when stuff related to a feature is together with the feature, not in a completely different file/bundle as suggested here.

Maintainability is debatable, I find the base code much more maintainable when it has the minimum amount of features required to work, not extra fields from all over the place.

But if the alternative is making a patch, then indeed it's not better. Patches are worse and should be removed.
For me the only good solution is to have a dedicated part of code (service/store/whatever the name) to store information contextual to the feature, together with the feature.

Message pin service is a good example, though we do lack some code to make it easier (to map extra information onto existing records in a way that satisfies all the above points with an elegant syntax).